src/fastfield/reader.rs

```diff
@@ -61,6 +62,34 @@ pub trait FastFieldReader<Item: FastValue>: Clone {
     fn max_value(&self) -> Item;
 }
 
+struct FFHeader {
+    field_id: u8,
+    gcd: u64,
+    min_value: u64,
+}
+
+fn read_header(bytes: &mut OwnedBytes) -> FFHeader {
+    let magic_number_or_field_id = bytes.read_u8();
+    if magic_number_or_field_id == FF_HEADER_MAGIC_NUMBER {
+        let _header_version = bytes.read_u8();
+        let field_id = bytes.read_u8();
+        let gcd = bytes.read_u64();
+        let min_value = bytes.read_u64();
+        FFHeader {
+            field_id,
+            gcd,
+            min_value,
+        }
+    } else {
+        // old version
+        FFHeader {
+            field_id: magic_number_or_field_id,
+            gcd: 1,
+            min_value: 0,
```

Comment:
can we make that header version so it does not byte us again?

Reply:
I don't understand, we have now a header version in the metadata that allow us to parse different header versions, but it's unused currently since there's only one header version.

Reply:
Ah that's the `header_version`? Ok good.